app/helpers/application_helper.rb

```diff
@@ -224,4 +224,19 @@ def render_initial_state
     content_tag(:script, json_escape(json).html_safe, id: 'initial-state', type: 'application/json')
     # rubocop:enable Rails/OutputSafety
   end
+
+  def grouped_scopes(scopes)
+    scope_parser      = ScopeParser.new
+    scope_transformer = ScopeTransformer.new
+
+    scopes.each_with_object({}) do |str, h|
+      scope = scope_transformer.apply(scope_parser.parse(str))
```

Comment:
Surely the scopes already do get parsed and you don't need to roll your own parser?

Reply:
No, they do not get parsed like I want. They are already an array of strings, but that’s not enough. We have a system for designating access type in the way our scopes are named. This allows me to group them better.

Reply:
I wonder if involving parslet isn't a little overkill, but it looks sound, and I understand why it is here.